lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Addresses.hs

```diff
@@ -253,6 +257,83 @@ spec = describe "SHELLEY_ADDRESSES" $ do
                 (`shouldNotSatisfy` T.isPrefixOf "addr_test")
             ]
 
+    it "ADDRESS_LIST_06 - Change addresses are listed even after transaction is no longer pending" $ \ctx -> runResourceT $ do
+        let initPoolGap = 10
+
+        let verifyAddrs i p addr = do
+                liftIO (length (filter ((== Used) . (^. (#state . #getApiT))) addr)
+                    `shouldBe` p)
+                liftIO (length addr `shouldBe` i)
+
+        -- 1. create shelley wallet
+        wSrc <- fixtureWallet ctx
+        wDest <- emptyWalletWith ctx ("Wallet", "cardano-wallet", initPoolGap)
+
+        -- 2. list addresses of that wallet (for later comparison)
+        listAddresses @n ctx wSrc >>= verifyAddrs 50 10
+
+        -- 3. send a transaction from that wallet
+        addrs <- listAddresses @n ctx wDest
+        verifyAddrs 20 0 addrs
+        let destination = (addrs !! 0) ^. #id
+        let payload = Json [json|{
+                "payments": [{
+                    "address": #{destination},
+                    "amount": {
+                        "quantity": #{2 * minUTxOValue},
+                        "unit": "lovelace"
+                    }
+                }],
+                "passphrase": "cardano-wallet"
+            }|]
+
+        request @(ApiTransaction n) ctx
+            (Link.createTransaction @'Shelley wSrc) Default payload
+            >>= flip verify
+                [ expectResponseCode HTTP.status202 ]
+
+        -- 4. wait for transaction to be inserted
+        eventually "Wallet balance = minUTxOValue" $ do
+            rb <- request @ApiWallet ctx
+                (Link.getWallet @'Shelley wDest) Default Empty
+            expectField
+                (#balance . #getApiT . #available)
+                (`shouldBe` Quantity (2 * minUTxOValue))
+                rb
+
+        -- 5. Check that change address is still there as unused
+        -- and the gap moved.
+        listAddresses @n ctx wSrc >>= verifyAddrs 51 11
+
+        -- 6. send another transaction from the destination wallet
+        listAddresses @n ctx wDest >>= verifyAddrs 21 1
```

Comment:
this is what should be expected after money was sent to  address with index = 0. This should be address from external pool. After receiving funds this address should have flipped to Used. And because we need `initAddressGap=10` so one Unused address extended external address pool. internal (aka change) address pool stays intact, ie. 10 Unused addresses